apps/src/templates/teacherDashboard/SectionActionBox.jsx

```diff
@@ -0,0 +1,216 @@
+import React, {Component, PropTypes} from 'react';
+import i18n from '@cdo/locale';
+import color from "../../util/color";
+import FontAwesome from '../FontAwesome';
+import PrintCertificates from "./PrintCertificates";
+import {connect} from 'react-redux';
+import {sectionCode,
+        sectionName,
+        removeSection,
+        toggleSectionHidden,
+        importOrUpdateRoster} from './teacherSectionsRedux';
+import {sortableSectionShape, OAuthSectionTypes} from "./shapes.jsx";
+import {pegasus} from "../../lib/util/urlHelpers";
+import * as utils from '../../utils';
+import BaseDialog from '../BaseDialog';
+import Button from '../Button';
+import DialogFooter from "./DialogFooter";
+
+
+const styles = {
+  arrowIcon: {
+    paddingRight: 8
+  },
+  actionBox: {
+    width: 180,
+    paddingLeft: 15,
+    paddingRight: 15,
+    border: '1px solid ' + color.lighter_gray,
+    borderRadius: 2,
+    backgroundColor: color.white,
+    boxShadow: "2px 2px 2px " + color.light_gray,
+    marginTop: 5,
+  },
+  actionText: {
+    fontSize: 14,
+    fontFamily: '"Gotham", sans-serif',
+    color: color.charcoal,
+    padding: '10px 0px 4px 0px',
+  },
+  actionTextBreak: {
+    fontSize: 14,
+    fontFamily: '"Gotham", sans-serif',
+    borderTop: '1px solid ' + color.lighter_gray,
+    color: color.charcoal,
+    padding: '13px 0px 2px 0px',
+    marginTop: '7px',
+  },
+  archiveDelete: {
+    color: color.red,
+    padding: '10px 0px 10px 0px',
+    fontSize: 14,
+  },
+  xIcon: {
+    paddingRight: 5,
+  },
+  bold: {
+    fontFamily: '"Gotham 5r", sans-serif'
+  },
+};
+
+class SectionActionBox extends Component {
+  static propTypes = {
+    style: PropTypes.object,
+    handleEdit: PropTypes.func,
+    sectionData: sortableSectionShape.isRequired,
+
+    //Provided by redux
+    removeSection: PropTypes.func.isRequired,
+    toggleSectionHidden: PropTypes.func.isRequired,
+    sectionCode: PropTypes.string,
+    sectionName: PropTypes.string,
+    updateRoster: PropTypes.func.isRequired,
+  };
+
+  state = {
+    deleting: false,
+  };
+
+  onConfirmDelete = () => {
+      const {removeSection } = this.props;
+      const section = this.props.sectionData;
+      $.ajax({
```

Comment:
We should handle the intermediate state while this request is in flight gracefully - at least disable the confirm delete button so we can't request this twice, possibly hide the dialog immediately but put the row in a 'pending' state until it gets removed, that sort of thing.